docs/spec2006-tutorial.md

```diff
@@ -0,0 +1,517 @@
+		 cd gem5
+		 git remote add darchr https://github.com/darchr/gem5
+	         git fetch darchr  
+	         git cherry-pick 6450aaa7ca9e3040fb9eecf69c51a01884ac370c  
+	         git cherry-pick 3403665994b55f664f4edfc9074650aaa7ddcd2c''',
+    typ = 'git repo',
+    name = 'gem5',
+    path =  'gem5/',
+    cwd = './',
+    documentation = 'cloned gem5 master branch from googlesource and cherry-picked 2 commits on Nov 20th'
+)
+
+gem5_binary = Artifact.registerArtifact(
+    command = 'scons build/X86/gem5.opt -j8',
+    typ = 'gem5 binary',
+    name = 'gem5',
+    cwd = 'gem5/',
+    path =  'gem5/build/X86/gem5.opt',
+    inputs = [gem5_repo,],
+    documentation = 'compiled gem5 binary right after downloading the source code'
+)
+```
+### Building m5  
+m5 is a binary that facilitates the communication between the host system and the guest system (gem5). 
+The examples of how m5 is used could be found in the runscripts that we will describe later. 
+m5 binary will be copied to the disk image, so that the guest could run m5 binary during the experiment. 
+Therefore, m5 binary should be compiled before we build the disk image.  
+
+To compile m5 binary,  
+```sh
+cd gem5/util/m5/
+make -f Makefile.x86
+```
+In launch_spec_experiments.py, we document the step in an Artifact object as follows,  
+```python
+m5_binary = Artifact.registerArtifact(
+    command = 'make -f Makefile.x86',
+    typ = 'binary',
+    name = 'm5',
+    path =  'gem5/util/m5/m5',
+    cwd = 'gem5/util/m5',
+    inputs = [gem5_repo,],
+    documentation = 'm5 utility'
+)
+```
+### Preparing Scripts to Modify the Disk Image
+In this step, we will prepare the scripts that will modify the disk image after the Ubuntu installation process has finished, and before the first time we use the disk image in gem5. 
+We will keep the related files in the disk-image folder of the experiment. 
+The files that are made specifically for SPEC 2006 benchmarks will be in disk-image/spec2006, and the files that are commonly used accross most benchmarks will be in disk-image/shared.  
+
+In the root folder of the experiment,
+```sh
+mkdir disk-image
+mkdir disk-image/spec2006
+mkdir disk-image/shared
+```
+
+The first script is the runscript.sh script, which will be appended to the end of .bashrc file. 
+Therefore, that script will run when we use the disk image in gem5, after the Linux booting process has finished. 
+Figure 1 describes how this script interacts with the gem5 config file.  
+The script could be found [here](https://github.com/darchr/gem5art/blob/master/docs/disks/spec2006/runscript.sh).
+
+To download the script, in the root folder of the experiment,
+```sh
+cd disk-image/spec2006
+wget https://raw.githubusercontent.com/darchr/gem5art/master/docs/disks/spec2006/runscript.sh
+```
+
+The second script is post-instsallation.sh script, which will copy the "auto logging in" script to the correct place, and copy the m5 binary to /sbin/ in the disk image. 
```

Comment:
For all filenames and paths, use ` like .bashrc above.

Also, installation is misspelled ;)